### PR TITLE
[linux] add --gl-interface=<interface> switch 

### PR DESCRIPTION
**Pull request by lrusak**
This adds a new command line switch for linux. The new switch controls the old `KODI_GL_INTERFACE` env variable. The purpose of which was to control the GLX or EGL selection (for X11 windowing only).

This also adds a wrapper in the `kodi.sh` script to notify the deprecation.

Example help text:
```
Usage: kodi [OPTION]... [FILE]...

Arguments:
  -fs                   Runs Kodi in full screen
  --standalone          Kodi runs in a stand alone environment without a window
                        manager and supporting applications. For example, that
                        enables network settings.
  -p or --portable      Kodi will look for configurations in install folder instead of ~/.kodi
  --debug               Enable debug logging
  --version             Print version information
  --test                Enable test mode. [FILE] required.
  --settings=<filename> Loads specified file after advancedsettings.xml replacing any settings specified
                        specified file must exist in special://xbmc/system/

Linux Specific Arguments:
  --windowing=<system>  Select which windowing method to use.
                          Available window systems are: wayland, gbm, x11
  --logging=<target>    Select which log target to use (log file will always be used in conjunction).
                          Available log targets are: console
  --audio-backend=<backend> Select which audio backend to use.
                          Available audio backends are: alsa, pipewire, pulseaudio, alsa+pulseaudio
  --gl-interface=<interface> Select which GL interface to use (X11 only).
                          Available GL interfaces are: glx, egl, egl-pb
```

### CODE REVIEW

**Review by wsnipex — APPROVED** [review]

**Review by fritsch — APPROVED** [review]
Thanks much!

File: cmake/modules/FindEGL.cmake [hunk 1]
```diff
@@ -29,6 +29,10 @@ if(NOT TARGET EGL::EGL)
                                     VERSION_VAR EGL_VERSION)
 
   if(EGL_FOUND)
+    list(APPEND GL_INTERFACES_LIST "egl")
+    list(APPEND GL_INTERFACES_LIST "egl-pb")
```
Comment (fuzzard) [inline cmake/modules/FindEGL.cmake:33]: you should just be able to do 
```
list(APPEND GL_INTERFACES_LIST egl egl-pb)
```
You can bracket each one if you want, or just add "egl-pb") on a new line like this if you prefer for distinction

```
    list(APPEND GL_INTERFACES_LIST "egl"
                                                                 "egl-pb")
```

Comment (lrusak) [reply cmake/modules/FindEGL.cmake]: updated

**Comment by lrusak** [comment]
squashed the fixup, should be good to go.

**Pull request merged**